src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs

```diff
@@ -3109,6 +3115,18 @@ private void AppendLineToOutputFunction(string name, string js, StringBuilder wr
                 return;
             }
 
+            if (js.StartsWith("include("))
+            {
+                Regex isDiscard = new Regex("_+", RegexOptions.IgnoreCase);
```

Comment:
This creates a new Regex each time. Create a static instance with `Compiled`, instead.

For that matter, why are you using a regex instead of a `Contains` or `IndexOf` ? 

What is this meant to catch?

Reply:
Done.
(Edited the Regex to a static instance with `Compiled`.)

We try catching what the var name given by the user when calling `RavenQuery.Include`

For example:
`var query = from u in session.Query<Order>()
                        let **includes** = RavenQuery.Include<Order>(u.Employee)
                        select new QueryResult { Comapny = u.Company };`

We found that it'll return an array of null values when we use `query.toList()`.
After consulting with Pawel, we decided to force the user to use '_' or '__', '___' etc... and throw when a user tries using something else.
So it meant to make sure the user made a valid query. I think `contains` and `IndexOf` are not suitable for it